spec/writer_spec.rb

```diff
@@ -0,0 +1,11 @@
+require "./lib/writer"
+require "mocks/mock_io"
+
+describe "writer" do
+	let (:writer) { writer = Writer.new(MockIO.new) }
```

Comment:
You don't need to assign within the let block just have it be:

`let(:writer) { Writer.new(MockIO.new) }`